### PR TITLE
Add support for stream outline nodes in the reader

### PR DESCRIPTION
**Pull request by adrg**
Addresses #57

---
This change is [Reviewable](https://reviewable.io/reviews/unidoc/unipdf/79)

### CODE REVIEW

File: model/reader.go [hunk 1]
```diff
@@ -282,12 +282,15 @@ func (r *PdfReader) loadOutlines() (*PdfOutlineTreeNode, error) {
 
 	outlineRoot, ok := outlineRootObj.(*core.PdfIndirectObject)
 	if !ok {
-		if _, ok := core.GetDict(outlineRootObj); !ok {
+		if _, ok := core.GetDict(outlineRootObj); ok {
+			common.Log.Debug("Outline root is a dict. Should be an indirect object")
+			outlineRoot = core.MakeIndirectObject(outlineRootObj)
+		} else if outlineStream, ok := core.GetStream(outlineRootObj); ok {
+			common.Log.Debug("Outline root is a stream. Should be an indirect object")
+			outlineRoot = core.MakeIndirectObject(outlineStream.PdfObjectDictionary)
```
Comment (gunnsth) [inline model/reader.go]: Does this actually fully make sense?  Same as simply ignoring this error?  Is there any case where the actual outlines object is a stream?  Did not see in the reference.  Checking some of those files in the issue seemed like the outlines are not working.  Preflight in Adobe says "The value associated with the key is of incorrect type" - Key: Outlines.
Might be more clear to just say "Invalid outline root - skipping"

Comment (adrg) [reply model/reader.go]: I think you are correct. The outlines in those files don't seem to be read by Adobe Reader. So skipping them should be ok. Should we do this whenever the outline root is anything invalid?

Comment (gunnsth) [reply model/reader.go]: Yes I think that would be good, just outputting a debug message.  Similar to ghostscript handling.

Comment (adrg) [reply model/reader.go]: I created another PR with the changes. This one can be closed.

**Pull request closed without merging**